src/app/app.component.ts

```diff
@@ -8,34 +8,60 @@ import {ActivatedRoute, ParamMap} from '@angular/router';
 })
 
 export class AppComponent implements OnInit {
-    public stackUrl: string; //'http://bayesian-api-bayesian-preview.b6ff.rh-idev.openshiftapps.com/api/v1/stack-analyses-v2/132389240e2342409dda9b8c800a905d';
-    // public stackUrl: string = 'http://localhost:32000/api/v1/stack-analyses/ed6fc94dbe63454093c8586e5bb811dd';
+    public stackUrl: string;
+    public apiData: any;
+    public gateway: any = {};
+    public label: string;
+    public routerLink: string;
 
     constructor(private route: ActivatedRoute) {
         this.route.paramMap.subscribe((params) => {
             this.label = params.get('id');
         });
         window.onhashchange = () => {
-            let id: string = location.hash.replace('#/analyze/', '');
-            this.label = id;
-            this.changeLabel();
+            let url: string = location.hash;
+            let id: string = url.replace('#/analyze/', '');
+            let splitParams: Array<string> = id.split('?');
+            this.label = splitParams[0];
+            this.onAppLoad();
         };
     }
 
-    public label: string;
-    public routerLink: string;
-    // d6819b27a4ba4e8fa6f6bf63bb7764ee;
-    changeLabel() {
+    onAppLoad(): void {
+        let url: string = location.hash;
+        let id: string = url.replace('#/analyze/', '');
+        let splitParams: Array<string> = id.split('?');
+        debugger;
```

Comment:
`debugger` can be removed , and `let url: string = location.hash; let id: string = url.replace('#/analyze/', '');` can be passed to function call `onAppLoad()` ??